recipes/mmdet/meta.yaml

```diff
@@ -0,0 +1,53 @@
+{% set name = "mmdet" %}
+{% set version = "2.25.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/open-mmlab/mmdetection/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 6642d593395a6c2d13bd1a574525477a8d08ef0fb7e67f9127a91b67193b06a9
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.7
+    - pip
+    - pytorch
+  run:
+    - python >=3.7
+    - numpy
+    - pycocotools
+    - six
+    - terminaltables
+
+test:
+  imports:
+    - mmdet
+  requires:
+    - mmcv-full
+    - pytorch
+    - opencv
```

Comment:
why are `opencv`, `pytorch` and `mmcv-full` not `run` requirements?

Reply:
upstream package doesn't depend on those  in wheel distribution:

```
mmdet==2.25.0
  - matplotlib [required: Any, installed: 3.5.2]
    - cycler [required: >=0.10, installed: 0.11.0]
    - fonttools [required: >=4.22.0, installed: 4.33.3]
    - kiwisolver [required: >=1.0.1, installed: 1.4.3]
    - numpy [required: >=1.17, installed: 1.23.0]
    - packaging [required: >=20.0, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pillow [required: >=6.2.0, installed: 7.0.0]
    - pyparsing [required: >=2.2.1, installed: 3.0.9]
    - python-dateutil [required: >=2.7, installed: 2.7.3]
  - numpy [required: Any, installed: 1.23.0]
  - pycocotools [required: Any, installed: 2.0.4]
    - matplotlib [required: >=2.1.0, installed: 3.5.2]
      - cycler [required: >=0.10, installed: 0.11.0]
      - fonttools [required: >=4.22.0, installed: 4.33.3]
      - kiwisolver [required: >=1.0.1, installed: 1.4.3]
      - numpy [required: >=1.17, installed: 1.23.0]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pillow [required: >=6.2.0, installed: 7.0.0]
      - pyparsing [required: >=2.2.1, installed: 3.0.9]
      - python-dateutil [required: >=2.7, installed: 2.7.3]
    - numpy [required: Any, installed: 1.23.0]
  - six [required: Any, installed: 1.14.0]
  - terminaltables [required: Any, installed: 3.1.10]
```

but yeah, I guess it's fair to add those since they directly do `import cv2` and `import torch` in some utility functions.

Reply:
I guess they don't add those because of the same reason we discussed the other day:
1. `opencv` has different package names that deliver same `cv2` package
2. `mmcv-full` vs `mmcv` - same
3. not sure about torch